File: Java/Newton.java
public class Newton {

    private static double f(double x){

        return Main.f(x);
    }

    private static double df(double x){

        return Main.df(x);
    }

    public static void Calcular(double xi, double tol, int maxIter){

        double x = xi;
        double error = tol + 1;
        double fx = f(x);
        double dfx;
        int iter = 0;

        System.out.printf("\n%4s %10s %10s %12s\n\n", "iter", "x", "f(x)", "Error");

        System.out.printf("|%4d| %10f| %10f| %10s|\n", iter, x, fx, "");

        while (error > tol && fx != 0 && iter < maxIter - 1) {

            dfx = df(x);
            x -= fx/dfx;
            fx = f(x);
            error = xi - x;
            error = Math.abs(error);
            xi = x;
            iter++;

            System.out.printf("|%4d| %10f| %10f| %10f|\n", iter, x, fx, error);
        }

        System.out.printf("\nLa raiz es de f(x) = x * exp(x) - x^2 - 5x - 3; es: %.15f\n", x);
    }
}

File: Entrega final/webSite/contenido/metodos/oneVariable/newton.py
from math import *


# err_type = abs | rel | fx
def newton(f, df, x0, tol, n, err_type):
    x = x0
    fx = f(x)
    dfx = df(x)
    actual_iterations = 0
    table = []

    error = float("inf")

    table.append(actual_iterations)
    table.append(x)
    table.append(fx)
    table.append(dfx)
    table.append("")
    table.append("")
    table.append("newline")

    if err_type == "fx":
        error = abs(fx)

    while fx != 0 and error > tol and dfx != 0 and actual_iterations < n:
        xn = x - (fx / dfx)
        fx = f(xn)
        dfx = df(xn)

        abs_err = abs(xn - x)
        rel_err = abs_err / abs(xn)

        if err_type == "fx":
            error = abs(fx)
        elif err_type == "rel":
            error = rel_err
        else:
            error = abs_err

        x = xn
        actual_iterations = actual_iterations + 1

        table.append(actual_iterations)
        table.append(x)
        table.append(fx)
        table.append(dfx)
        table.append(abs_err)
        table.append(rel_err)
        table.append("newline")

    if fx == 0:
        table.append("nan")
        return x, table
    elif error < tol:
        return x, table
    else:
        table.append("nan")
        return None, table
